### PR TITLE
Add portmap support

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add support for port mappings in CNI networking scenarios.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
This now contains unit and integration tests and is ready for review. Please note that the API is public and could be used by other projects in an Independent way.

PTAL @mrunalp

**Review by giuseppe — APPROVED** [review]
a small nit

LGTM

**Comment by saschagrunert** [comment]
> a small nit
> 
> LGTM

Where is this small nit? :grin:

**Comment by mrunalp** [comment]
Sorry for delay. I will mkae a pass at the open PRs tomorrow.

**Comment by saschagrunert** [comment]
> Sorry for delay. I will mkae a pass at the open PRs tomorrow.

Thank you! :pray:

File: src/network/cni/iptables.rs [hunk 1]
```diff
@@ -0,0 +1,1177 @@
...
+        f.debug_struct("Sandbox")
+            .field("iptables_binary", self.iptables_binary())
+            .field("ip6tables_binary", self.ip6tables_binary())
+            .field("use_ipv6", &self.use_ipv6())
+            .finish()
+    }
+}
+
+#[async_trait]
+trait Exec: DynClone + Send + Sync {
```
Comment (mrunalp) [inline src/network/cni/iptables.rs:89]: nit: Call this `ExecCommand` ?

File: src/network/cni/iptables.rs [hunk 2]
```diff
@@ -0,0 +1,1177 @@
...
+impl Iptables for DefaultIptables {
+    async fn setup(&self, chain: &Chain) -> Result<()> {
+        self.chain_setup(chain).await
+    }
+
+    async fn teardown(&self, chain: &Chain) -> Result<()> {
+        self.chain_teardown(chain).await
+    }
+
+    fn set_ipv6(&mut self, value: bool) {
```
Comment (mrunalp) [inline src/network/cni/iptables.rs:73]: Is this something that should be bool as part of `setup`?

File: src/network/cni/iptables.rs (hunk 1 above)
Comment (saschagrunert) [reply src/network/cni/iptables.rs]: Fixed as suggested :+1:

File: src/network/cni/iptables.rs (hunk 2 above)
Comment (saschagrunert) [reply src/network/cni/iptables.rs:73]: I thought about it, and we would also need it for `teardown`. Then we would have to pass around the value down to `run`, which seems a bit overkill to me. My main reason to add this method was to not hat a need to lookup the binary at every invocation.

**Comment by saschagrunert** [comment]
Unit tests are failing because of the nightly toolchain which is broken right now. Not related to this PR.

**Comment by saschagrunert** [comment]
Had to rebase, PTAL @mrunalp

**Comment by saschagrunert** [comment]
PTAL @mrunalp

**Pull request merged**